CMakeLists.txt

```diff
@@ -47,6 +50,58 @@ include_directories(lobster/include lobster/src)
 add_executable(treesheets ${sources})
 
 target_link_libraries(treesheets PRIVATE ${wxWidgets_LIBRARIES})
+if (NOT APPLE AND NOT WIN32 AND NOT TREESHEETS_RELOCATABLE_INSTALLATION)
+    # Apple and Windows always look to relative paths for locales.
+    target_compile_definitions(treesheets PRIVATE "LOCALEDIR=L\"${CMAKE_INSTALL_FULL_LOCALEDIR}\"")
+endif()
+
+install(TARGETS treesheets DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(DIRECTORY TS/docs/ DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(FILES TS/readme.html DESTINATION ${CMAKE_INSTALL_PREFIX})
+
+install(DIRECTORY TS/examples DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(DIRECTORY TS/images DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(DIRECTORY TS/images DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(FILES TS/images/treesheets.svg DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/scalable/apps)
+install(DIRECTORY TS/scripts DESTINATION ${CMAKE_INSTALL_PREFIX})
 
-install(TARGETS treesheets DESTINATION .)
-install(DIRECTORY TS/ DESTINATION .)
+# Install translations to correct platform-specific path.
+# See: https://docs.wxwidgets.org/trunk/overview_i18n.html#overview_i18n_mofiles
+file(
+    GLOB locales
+    RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}/TS/translations
+    TS/translations/*/ts.mo
+)
+list(
+    TRANSFORM locales
+    REPLACE "/ts\\.mo$" ""
+)
+if (APPLE)
+    foreach(locale ${locales})
+        install(
+            FILES "TS/translations/${locale}/ts.mo"
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/ThreeSheets.app/Contents/Resources/${locale}.lproj"
+        )
+    endforeach()
+elseif (WIN32)
+    foreach(locale ${locales})
+        install(
+            FILES "TS/translations/${locale}/ts.mo"
```

Comment:
I am not sure what this is going to do. Even if we ever had an "installer creation" on Windows, there's no need to be copying locale files, these files are tiny and the right one is automatically picked up by treesheets anyway. Better to just assume the files will get copied wholesale, and not introduce things which may break in interesting ways in the future.

Reply:
Same for os x

Reply:
This will copy the locales to the correct path according to the platform, as described by the [wxWidgets docs](https://docs.wxwidgets.org/trunk/overview_i18n.html#overview_i18n_mofiles).

I also want to make msgfmt call part of the build as [`.mo` files are not portable](https://www.gnu.org/software/gettext/manual/html_node/Files.html). Possibly, that can cause issues on systems e.g. with different endianness.

Reply:
That just shows where they would reside relative to the executable on win/osx. In both cases, they'd be in an installer or app package along with all the other standard files, and TreeSheets already knows how to access those, so there is no need for an install action on these platforms.

Reply:
I am not very familiar with those platforms, but I would expect CMake to generate installers/bundles for those platforms (e.g. https://cmake.org/cmake/help/latest/cpack_gen/nsis.html for Windows). So if we want to use CMake on those platforms, we will also want to specify the platform-specific paths here.

Reply:
if we ever do so thru CMake, we will do the entire data dir, not just individual locale files. Please remove.